src/App.jsx

```diff
@@ -1,20 +1,49 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import "./App.css";
 import Form from "./components/Form/Form.jsx";
 import { uid } from "uid";
 import List from "./components/List/List.jsx";
 
 function App() {
   const [activities, setActivities] = useState([]);
+  const [data, setData] = useState(null);
 
   function handleAddActivity(newActivity) {
     setActivities([...activities, { id: uid(), ...newActivity }]);
   }
-  console.log(activities);
+
+  const filteredActivities = activities.filter(
+    (activity) => activity.isForGoodWeather === data.isGoodWeather
+  );
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(
+          "https://example-apis.vercel.app/api/weather/europe"
+        );
+        if (!response.ok) {
+          throw new Error("Network error");
+        }
+
+        const result = await response.json();
+        setData(result);
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
```

Comment:
- Please avoid using arrow functions, but rather use traditional function declaration styles to maintain consistency in your code base. 🧽 
- please use a more descriptive function name, for better readably.
- Great that you are using error handling. The proposed approach for **error handling** has some advantages. It provides a more comprehensive and clear handling of both **network and data processing errors**, improves readability and maintainability of the code, and offers better flexibility for error logging and handling.
_But that's complaining at a high level._

```suggestion
    async function fetchWeather() {
      try {
        const response = await fetch('https://example-apis.vercel.app/api/weather/rainforest');

        if (response.ok) {
          const weatherData = await response.json();

          setWeather(weatherData);
        } else {
          console.error('bad server response');
        }
      } catch (error) {
        console.error(error);
      }
    };
```